### PR TITLE
Adding Resource parameter to AuthorizeRouteView

### PR DESCRIPTION
**Pull request by slorello89**
Summary of the changes (Less than 80 chars)
 - Adding Resource parameter to `AuthorizeRouteView`

Addresses #22679 (in this specific format)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=24035) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=24035) before we can accept your contribution.<br/><br/>:x: slorello89 [sign now](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=24035)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/aspnetcore?pullRequest=24035) it.</sub>

**Comment by slorello89** [comment]
(adding this hear so as not to break the 80 char limit)

I've added tests that appear in line with the unit testing done on this project beforehand. I was able to create a fairly contrived example of it's usage in the TestServer, I am not submitting that as it did not play nicely with [AuthorizeViewCases.razor](https://github.com/dotnet/aspnetcore/blob/master/src/Components/test/testassets/BasicTestApp/AuthTest/AuthorizeViewCases.razor) as that component is a child of [AuthRouter.razor](https://github.com/dotnet/aspnetcore/blob/master/src/Components/test/testassets/BasicTestApp/AuthTest/AuthRouter.razor) - and the Resource does not cascade down to the children. If you would like an example added to the TestServer let me know - I did not expend the time getting it working in the full context of that app as the AuthorizeView's Resource parameter was similarly not tested.

This is my first time contributing and while I did my beds to make sure I followed all the processes I apologize if I missed anything - feel free to send this back as a needs work if so :). Thank you!

File: src/Components/Authorization/src/AuthorizeRouteView.cs [hunk 1]
```diff
@@ -54,6 +54,12 @@ public AuthorizeRouteView()
         [Parameter]
         public RenderFragment Authorizing { get; set; }
 
+        /// <summary>
+        /// The resource to which access is being controlled.
+        /// </summary>
+        [Parameter]
+        public object Resource { get; set; }
```
Comment (captainsafia) [inline src/Components/Authorization/src/AuthorizeRouteView.cs:61]: Nice! You'll probably need to regenerate the reference assemblies for this.

You can do that by running `dotnet msbuild /t:GenerateReferenceSource` in the `src/Components/Authorization` directory.

File: src/Components/Authorization/test/AuthorizeRouteViewTest.cs [hunk 2]
```diff
@@ -76,6 +76,74 @@ public void WhenAuthorized_RendersPageInsideLayout()
                 edit => AssertPrependText(batch, edit, "Hello from the page with message: Hello, world!"));
         }
 
+        [Fact]
```
Comment (captainsafia) [inline src/Components/Authorization/test/AuthorizeRouteViewTest.cs:79]: The formatting on this file is a little off. Try running the formatter in VS.

If you're developing in VS Code, I've had a lot of success with the `C# FixFormat Fixed` extension.

**Review by captainsafia — COMMENTED** [review]
Looks good! Left a few notes in-line.

File: src/Components/Authorization/src/AuthorizeRouteView.cs (hunk 1 above)
Comment (slorello89) [reply src/Components/Authorization/src/AuthorizeRouteView.cs:61]: I've regenerated from `src/Components/Authorization`

File: src/Components/Authorization/test/AuthorizeRouteViewTest.cs (hunk 2 above)
Comment (slorello89) [reply src/Components/Authorization/test/AuthorizeRouteViewTest.cs:79]: Think I see the formatting issues you're talking about - I moved the brace off of the Method Definition line

File: src/Components/Authorization/test/AuthorizeRouteViewTest.cs [hunk 3]
```diff
@@ -76,6 +76,77 @@ public void WhenAuthorized_RendersPageInsideLayout()
...
+                },
+                edit => AssertPrependText(batch, edit, "Layout ends here"));
+
+            // Assert: renders page
+            var pageDiff = batch.GetComponentDiffs<TestPageRequiringAuthorization>().Single();
+            Assert.Collection(pageDiff.Edits,
+                edit => AssertPrependText(batch, edit, "Hello from the page with message: Hello, world!"));
+
+            // Assert: Asserts that the Resource is present and set to "foo"
+            Assert.Collection(_testAuthorizationService.AuthorizeCalls, call=>{
```
Comment (pranavkm) [inline src/Components/Authorization/test/AuthorizeRouteViewTest.cs:116]:
```suggestion
            Assert.Collection(_testAuthorizationService.AuthorizeCalls, call=>
            {
```

File: src/Components/Authorization/test/AuthorizeRouteViewTest.cs [hunk 4]
```diff
@@ -76,6 +76,77 @@ public void WhenAuthorized_RendersPageInsideLayout()
...
+            // Assert: renders layout containing "not authorized" message
+            var batch = _renderer.Batches.Single();
+            var layoutDiff = batch.GetComponentDiffs<TestLayout>().Single();
+            Assert.Collection(layoutDiff.Edits,
+                edit => AssertPrependText(batch, edit, "Layout starts here"),
+                edit => AssertPrependText(batch, edit, "Not authorized"),
+                edit => AssertPrependText(batch, edit, "Layout ends here"));
+
+            // Assert: Asserts that the Resource is Null
+            Assert.Collection(_testAuthorizationService.AuthorizeCalls, call=>{
```
Comment (pranavkm) [inline src/Components/Authorization/test/AuthorizeRouteViewTest.cs:145]:
```suggestion
            Assert.Collection(_testAuthorizationService.AuthorizeCalls, call=>
            {
```

File: src/Components/Authorization/test/AuthorizeRouteViewTest.cs [hunk 5]
```diff
@@ -76,6 +76,77 @@ public void WhenAuthorized_RendersPageInsideLayout()
...
+                edit => AssertPrependText(batch, edit, "Layout ends here"));
+
+            // Assert: renders page
+            var pageDiff = batch.GetComponentDiffs<TestPageRequiringAuthorization>().Single();
+            Assert.Collection(pageDiff.Edits,
+                edit => AssertPrependText(batch, edit, "Hello from the page with message: Hello, world!"));
+
+            // Assert: Asserts that the Resource is present and set to "foo"
+            Assert.Collection(_testAuthorizationService.AuthorizeCalls, call=>{
+                Assert.Equal(resource,call.resource.ToString());
```
Comment (pranavkm) [inline src/Components/Authorization/test/AuthorizeRouteViewTest.cs:117]:
```suggestion
                Assert.Equal(resource, call.resource.ToString());
```

File: src/Components/Authorization/test/AuthorizeRouteViewTest.cs [hunk 6]
```diff
@@ -76,6 +76,77 @@ public void WhenAuthorized_RendersPageInsideLayout()
...
+        }
+
+        [Fact]
+        public void NotAuthorizedWhenResourceMissing()
+        {
+            // Arrange
+            var routeData = new RouteData(typeof(TestPageRequiringAuthorization), EmptyParametersDictionary);
+            _testAuthorizationService.NextResult = AuthorizationResult.Failed();
+
+
```
Comment (pranavkm) [inline src/Components/Authorization/test/AuthorizeRouteViewTest.cs:128]: Could you remove this extra line

**Review by pranavkm — APPROVED** [review]
@javiercn does this look good to you?

File: src/Components/Authorization/test/AuthorizeRouteViewTest.cs (hunk 6 above)
Comment (slorello89) [reply src/Components/Authorization/test/AuthorizeRouteViewTest.cs]: Done 👍

**Review by javiercn — APPROVED** [review]

**Comment by javiercn** [comment]
@slorello89 thanks for the contribution!

**Pull request merged**